pkg/providers/imagefamily/resolver.go

```diff
@@ -111,6 +90,19 @@ func (r Resolver) Resolve(ctx context.Context, nodeClass *v1alpha2.AKSNodeClass,
 	return template, nil
 }
 
+func prepareKubeletConfiguration(instanceType *cloudprovider.InstanceType, nc *corev1beta1.NodeClaim) *corev1beta1.KubeletConfiguration {
+	kubeletConfig := nc.Spec.Kubelet
+	if kubeletConfig == nil {
+		kubeletConfig = &corev1beta1.KubeletConfiguration{}
+	}
+	kubeletConfig.MaxPods = lo.ToPtr[int32](consts.DefaultKubernetesMaxPodsAzure)
```

Comment:
`consts.DefaultKubernetesMaxPodsAzure' is a fixed value, and some nodes may not seem to need to allocate so many IPs. Can we first obtain it through the maxpod field in the nodepool, and only use the default value if it is not defined?

Reply:
kubelet configuration isnt something we are supporting for GA

Reply:
Okay, thank you for your answer

Reply:
Yeah, that was my same rationale, we should just do the max until we support setting it. The default for azure cni without overlay is 30, but for azure cni with overlay is 250. 

That also makes me wonder if that limit comes from the pod ip scaling of overlay and maybe there is a historical reason for that limit.

Reply:
Discussed offline. There is a concern about exhausting the IP space quickly; on the other hand Karpenter can provision larger nodes more often, so underspecifying max-pods is also undesirable. Keeping it 250 for now, will revisit (hopefully just make it configurable).